File: Day 3/1 Find Substring/solution.cs
using System.Globalization;

namespace Strings_Find_Word
{
    internal class Program
    {
        static void Main(string[] args)
        {
            string input = "Find the awesomes words in this string";
            string substringToFind = " awesome";

            int wordStartIndex = -1;
            int wordLength = substringToFind.Length;

            for (int i = 0; i < input.Length - wordLength + 1; i++)
            {
                if (input[i] == substringToFind[0])
                {
                    int j;
                    int foundChars = 0;

                    for (j = i; j < i + wordLength; j++)
                    {
                        if (input[j] != substringToFind[j - i])
                        {
                            break;
                        }

                        foundChars++;
                    }

                    if (foundChars == wordLength)
                    {
                        wordStartIndex = i;
                        break;
                    }
                }
            }

            if (wordStartIndex != -1)
            {
                Console.WriteLine($"Found '{substringToFind}' at index {wordStartIndex}.");
            }
            else
            {
                Console.WriteLine($"Could not find '{substringToFind}' in the input.");
            }


        }
    }
}

File: Day 3/1 Find Substring/solution.py
def find_word(input_str, word_to_find):
    word_length = len(word_to_find)
    word_start_index = -1

    for i in range(len(input_str) - word_length + 1):
        if input_str[i] == word_to_find[0]:
            j = i
            found_chars = 0

            while j < i + word_length and input_str[j] == word_to_find[j - i]:
                found_chars += 1
                j += 1

            if found_chars == word_length:
                word_start_index = i
                break

    if word_start_index != -1:
        print(f"Found '{word_to_find}' at index {word_start_index}.")
    else:
        print(f"Could not find '{word_to_find}' in the input.")
